microservices.yml

```diff
@@ -451,37 +461,127 @@ commands:
     description: Publish all unit test results to Jira Xray
     steps:
+      - attach_workspace:
+          at: .
       - run:
-          name: Download publish to Xray script
+          name: Upload Mocha unit test results to Jira Xray
           when: always
           command: |
-            # get push-test-results-to-xray.sh
-             wget https://raw.githubusercontent.com/MGMDV-Orbs/microservices/f/xray-r2.3/.circleci/push-test-results-to-xray.sh -O push-test-results-to-xray.sh
-      - run:
-          name: Upload Mocha unit test results to Jira Xray
+            TEST_RESULTS_FILE=./reports/unit-test-mocha-results.xml
+            PROJECT_CODE=SRV
+            TEST_TYPE=junit
+            echo "Push Xray Version 1.0.1"
+            echo "File to process: $TEST_RESULTS_FILE"
+            echo "Jira Project Name: $PROJECT_CODE"
+            echo "Test Type: $TEST_TYPE"
+            echo "CIRCLE_SHA1 = $CIRCLE_SHA1"
+            if ! test -f $TEST_RESULTS_FILE; then
+              echo "mocha test results file not found ($TEST_RESULTS_FILE)"
+              exit 1
+            fi
+            XRAY_AUTH_TOKEN=$(\
+              curl -s https://xray.cloud.xpand-it.com/api/v1/authenticate \
+                -H "Content-Type: application/json" -X POST \
+                --data "{ \"client_id\": \"$XRAY_CLIENT_ID\", \"client_secret\": \"$XRAY_CLIENT_SECRET\" }" \
+                | tr -d '"'
+            )
+            RESULTS_TEST=$(\curl "https://xray.cloud.xpand-it.com/api/v1/import/execution/junit?projectKey=$PROJECT_CODE&revision=$CIRCLE_SHA1" \
+              -H "Content-Type: text/xml" \
+              -X POST \
+              -H "Authorization: Bearer $XRAY_AUTH_TOKEN" \
+              --data @"$TEST_RESULTS_FILE"
+            )
+            echo "UPLOAD RESULT = $RESULTS_TEST"
+      - run:
+          name: Upload Cucumber unit test results to Jira Xray
           when: always
           command: |
-            # run script to push results to Jira
-            bash /home/circleci/project/push-test-results-to-xray.sh ./reports/unit-test-mocha-results.xml SRV junit
-
-  publish-integration-test-results-to-xray:
+            TEST_RESULTS_FILE=./reports/unit-test-results.json
+            TEST_TYPE=cucumber
+            echo "Push Xray Version 1.0.1"
+            echo "File to process: $TEST_RESULTS_FILE"
+            echo "Test Type: $TEST_TYPE"
+            if ! test -f $TEST_RESULTS_FILE; then
+              echo "cucumber test results file not found ($TEST_RESULTS_FILE)"
+              exit 1
+            fi
+            XRAY_AUTH_TOKEN=$(\
+              curl -s https://xray.cloud.xpand-it.com/api/v1/authenticate \
+                -H "Content-Type: application/json" -X POST \
+                --data "{ \"client_id\": \"$XRAY_CLIENT_ID\", \"client_secret\": \"$XRAY_CLIENT_SECRET\" }" \
+                | tr -d '"'
+            )
+            RESULTS_TEST=$(\curl "https://xray.cloud.xpand-it.com/api/v1/import/execution/cucumber" \
```

Comment:
Is there any need to assign this command's output to a variable?

Seems like the only thing we do with it is `echo` it to stdout. 
I believe we can just skip that by just not assigning output to the variable

Reply:
Also, do we know if a failing call from the subcommand will fail the circleci step properly?

Reply:
The check for the file will fail the circleci step. A failed post to xray to integrate the results will not fail the circleci step.